docs/user/AggFunc.md

```diff
@@ -0,0 +1,112 @@
+# Aggregate Function
+Aggregate functions output a single result value from a collection of input values.
+
+Unlike SQL, `GROUP BY` is not required to use the aggregate function. 
+
+Unless otherwise specified, aggregate functions will skip `NULL` and `MISSING`
+
+## AVG
+
+Returns the average (arithmetic mean) of the input expression values.
+
+Signature
+: `AVG: COLLECTION OF NUMERIC TYPE -> DECIMAL OR NULL`
+
+Header
+: `AVG(expr)`
+
+Purpose
+: Given a collection of numeric type values, evaluates the arithmetic mean of the input expression values. 
+
+Example
+: 
+
+```SQL
+    AVG(<< 1,2,3 >>) -- 2 (Decimal type with Precision of 1 and Scale of 0)
+    AVG(<< 1,2,3,'1'>>) --error
+    AVG(<<MISSING>>) -- NULL
+    AVG(<< 1,2,3, MISSING >>) -- 2
```

Comment:
Ideally yes, but I hesitate to designate this doc to partiql-docs at this point because I felt what we have implemented in the Lang-Kotlin may not be exactly what we desire for spec. 

For example, one may argue that `AVG(<< 1, 2, 3 >>` should return 2 as an integer instead of `decimal(1,0)`. 

My original intention for this document is to keep track of the current Lang-Kotlin behavior so that 1) customers has a place to refer to 2) we can cross reference the current implementation with sql spec and other popular sql implementations.

Reply:
Maybe the drafts folder under partiql-docs?

Reply:
Understood. It seems then that the aggregate functions need to be formally defined along with their conformance to the SQL specification